src/firefly/js/visualize/VisMouseSync.js

```diff
@@ -0,0 +1,81 @@
+
+
+
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import Enum from 'enum';
+import {primePlot} from './PlotViewUtil.js';
+import {visRoot} from './ImagePlotCntlr.js';
+import CsysConverter from './CsysConverter.js';
+
+export const MouseState= new Enum(['NONE', 'ENTER', 'EXIT', 'DOWN', 'UP',
+    'DRAG_COMPONENT', 'DRAG', 'MOVE', 'CLICK',
+    'DOUBLE_CLICK']);
+
+var lastCtx = {
+    mouseState : MouseState.NONE,
+    plotId : null,
+    screenPt : null,
+    imagePt : null,
+    worldPt : null,
+    modifiers: {}
+};
+
+var listenerList= [];
+var oneTimeCallList= [];
+
+
+export function lastMouseCtx() { return lastCtx; }
+
+export function addMouseListener(l) {
+    listenerList.push(l);
+    return () => {
+        const idx= listenerList.indexOf(l);
+        if (idx>-1) listenerList.splice(idx,1);
+    };
+}
+
+
+export function addOneTimeMouseCall(l) { oneTimeCallList.push(l); }
+
+export function mouseUpdatePromise() {
+    return new Promise((resolve) => addOneTimeMouseCall( (s) => resolve(s)) );
+}
+
+export function fireMouseCtxChange(mouseCtx) {
+    lastCtx= mouseCtx;
+    listenerList.forEach((l) => l(mouseCtx));
+    oneTimeCallList.forEach((l) => l(mouseCtx));
+    oneTimeCallList=[];
+}
+
```

Comment:
That is true sort of.  They is a new controller that holds the readout state.  It could get updated a lot.  The VisMosueState turned out to be more a way to listen to events instead of state.